shell/browser/extensions/api/resources_private/resources_private_api.cc

```diff
@@ -0,0 +1,112 @@
+// Copyright 2015 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "shell/browser/extensions/api/resources_private/resources_private_api.h"
+
+#include <memory>
+#include <string>
+#include <utility>
+
+#include "base/values.h"
+#include "chrome/browser/browser_process.h"
+#include "chrome/common/extensions/api/resources_private.h"
+#include "chrome/grit/generated_resources.h"
+#include "components/strings/grit/components_strings.h"
+#include "components/zoom/page_zoom_constants.h"
+#include "pdf/buildflags.h"
+#include "printing/buildflags/buildflags.h"
+#include "ui/base/l10n/l10n_util.h"
+#include "ui/base/webui/web_ui_util.h"
+
+#if BUILDFLAG(ENABLE_PDF)
+#include "pdf/pdf_features.h"
+#endif  // BUILDFLAG(ENABLE_PDF)
+
+// To add a new component to this API, simply:
+// 1. Add your component to the Component enum in
+//      chrome/common/extensions/api/resources_private.idl
+// 2. Create an AddStringsForMyComponent(base::DictionaryValue * dict) method.
+// 3. Tie in that method to the switch statement in Run()
+
+namespace extensions {
+
+namespace {
+
+void AddStringsForPdf(base::DictionaryValue* dict) {
+#if BUILDFLAG(ENABLE_PDF)
+  static constexpr webui::LocalizedString kPdfResources[] = {
```

Comment:
Have we disabled the `Print` feature in the pdf UI, as I don't think we can provide the capability at the moment ?

Reply:
somewhat related question, how will the Download feature in the pdf UI be supported?

Reply:
disabled it.

i'd like the download feature to work the same as other downloads, but it currently just does nothing.